xgi/algorithms/centrality.py

```diff
@@ -307,3 +308,75 @@ def line_vector_centrality(H):
             vc[node].append(c_i[node])
 
     return vc
+
+
+def katz_centrality(H, index=False, cutoff=100):
+    """Returns the Katz-centrality vector of hypergraph H.
+
+    Parameters
+    ----------
+    H : xgi.Hypergraph
+        Hypergraph on which to compute the alpha centralities.
+    index : bool
+        If set to True, will return a dictionary mapping each vector index to a node.
+    cutoff : int
+        Power at which to stop the computation A + alpha * A**2 + alpha**2 * A**3 + ...
+        Default is 100.
+
+    Returns
+    -------
+    c : np.ndarray
+        Vector of the node centralities, sorted by the node indexes.
+    nodedict : dict
+        If index is set to True, nodedict will contain the nodes ids, keyed by their indice in vector c.
+        <c[key]> will be the centrality of node <nodedict[key]>.
+
+    Note
+    ----
+    [1] The Katz-centrality is defined as :
+        c = [(I - alpha.A^{t})^{-1} - I] • (1, 1, ..., 1)
+    Where A is the adjency matrix of the the (hyper)graph.
+    Since A^{t} = A for undirected graphs (our case), we have :
+        (I + A + alpha * A**2 + alpha**2 * A**3 + ...) * (I - alpha.A^{t}) = (I + A + alpha * A**2 + alpha**2 * A**3 + ...) * (I - alpha.A)
+                                                                           = (I + A + alpha * A**2 + alpha**2 * A**3 + ...) - A - alpha * A**2 - alpha**2 * A**3 - alpha**3 * A**3 - ...
+                                                                           = I
+    And (I - alpha.A^{t})^{-1} = I + A + alpha * A**2 + alpha**2 * A**3 + ...
+    Thus we can use the power serie to compute the Katz-centrality.
+    [2] The Katz-centrality of isolated nodes (i.e. no hyperedges) is not clearly define.
+    The output in this case could probably be improved.
```

Comment:
What is the current output for the case without hyperedges?

```suggestion
    [2] The Katz-centrality of isolated nodes (i.e. no hyperedges) is not clearly defined.
    The output in this case could probably be improved.
```

Reply:
If the output is not defined, I think NaN or an error should be returned.

Reply:
Actually, the output should be zero for isolated nodes. I modified the corresponding code.
For an empty hypergraph, I decided to return an empty Numpy array....

Reply:
Also, now the function will raise an Error for empty hypergraphs (as the centrality is defined on nodes...).